frontend/packages/dev-console/integration-tests/support/step-definitions/monitoring/topology-sidebar-actions.ts

```diff
@@ -0,0 +1,190 @@
+When('user selects {string} from topology sidebar Actions dropdown', (menuOption: string) => {
+  topologySidePane.selectNodeAction(menuOption);
+});
+
+When('user clicks on the workload {string} to open the sidebar', (nodeName: string) => {
+  topologyPage.clickOnNode(nodeName);
+  topologySidePane.verify();
+});
+
+When(
+  'user clicks on the knative service {string} to open the sidebar',
+  (knativeWorkloadName: string) => {
+    topologyPage.clickOnKnativeService(knativeWorkloadName);
+  },
+);
+
+Then('page redirected to the Monitoring page', () => {
+  detailsPage.titleShouldContain('Monitoring');
+});
+
+Given('user is on the topology sidebar of the helm release {string}', (helmReleaseName: string) => {
+  createHelmChartFromAddPage(helmReleaseName);
+  topologyPage.clickOnHelmWorkload();
+});
+
+When('user clicks on From Git card', () => {
+  addPage.selectCardFromOptions(addOptions.Git);
+});
+
+When('user clicks on Add Readiness Probe', () => {
+  addHealthChecksPage.clickProbeLink('Add Readiness Probe');
+});
+
+When('user clicks on tick button', () => {
+  addHealthChecksPage.clickCheckIcon();
+});
+
+When('user clicks on Add Liveness Probe', () => {
+  addHealthChecksPage.clickProbeLink('Add Liveness Probe');
+});
+
+When('user clicks on Add Startup Probe', () => {
+  addHealthChecksPage.clickProbeLink('Add Startup Probe');
+});
+
+When('user clicks on Add button', () => {
+  addHealthChecksPage.clickAdd();
+});
+
+When(
+  'user right clicks on the workload {string} to open the Context Menu',
+  (workloadName: string) => {
+    topologyPage.rightClickOnNode(workloadName);
+  },
+);
+
+When(
+  'user right clicks on the Service {string} to open the Context Menu',
+  (serviceName: string) => {
+    topologyPage.rightClickOnKnativeService(serviceName);
+  },
+);
+
+When('user clicks on Edit Health Checks', () => {
+  topologyPage.selectContextMenuAction(nodeActions.EditHealthChecks);
+});
+
+When('user sees Readiness Probe already added', () => {
+  addHealthChecksPage.verifySuccessText('Readiness Probe Added');
+});
+
+When('user removes Readiness Probe', () => {
+  addHealthChecksPage.removeReadinessProbe();
+});
+
+When('user clicks on Save button', () => {
+  addHealthChecksPage.clickSave();
+});
+
+When('user right clicks on the {string} to open the Context Menu', (nodeName: string) => {
+  topologyPage.rightClickOnNode(nodeName);
+});
+
+Then('user will be taken to Dashboard tab on the Monitoring page', () => {
+  detailsPage.titleShouldContain(pageTitle.Monitoring);
+});
+
+Then('user wont see Monitoring tab', () => {
+  topologySidePane.verify();
+  cy.get(topologyPO.sidePane.tabName)
+    .contains(sideBarTabs.monitoring)
+    .should('not.be.visible');
+});
+
+Then('user will see View Monitoring dashborad link', () => {
+  cy.get(topologyPO.sidePane.monitoringTab.viewMonitoringDashBoardLink).should('be.visible');
+});
+
+Then('user will see CPU Usage Metrics', () => {
+  monitoringPage.dashboard.verifyCpuUsageGraph();
+});
```

Comment:
sorry i could not map this `then` with `when` i.e on which action this will happen?

Reply:
Given - Precondition
When - performs action
Then - verify the outcome